error.go

```diff
@@ -18,68 +18,85 @@ package edgedb
 
 import (
 	"errors"
-	"fmt"
-	"log"
 
 	"github.com/edgedb/edgedb-go/internal/buff"
 )
 
 var (
-	// Error is wrapped by all edgedb errors.
-	Error error = errors.New("")
+	// ErrZeroResults is returned when a query has no results.
+	ErrZeroResults error = &Error{&baseError{msg: "edgedb: zero results"}}
+)
 
-	// ErrReleasedTwice is returned if a PoolConn is released more than once.
-	ErrReleasedTwice = fmt.Errorf(
-		"connection released more than once%w", Error,
-	)
+func wrapError(err error) error {
+	if err == nil {
+		return nil
+	}
 
-	// ErrZeroResults is returned when a query has no results.
-	ErrZeroResults = fmt.Errorf("zero results%w", Error)
+	return &Error{&baseError{msg: "edgedb: " + err.Error(), err: err}}
+}
 
-	// ErrPoolClosed is returned by operations on closed pools.
-	ErrPoolClosed error = fmt.Errorf("pool closed%w", Error)
+// newErrorFromCode returns a new edgedb error.
+func newErrorFromCode(code uint32, msg string) error {
+	err := &Error{&baseError{msg: msg}}
+	return wrapErrorWithType(code, err)
+}
 
-	// ErrContextExpired is returned when an expired context is used.
-	ErrContextExpired error = fmt.Errorf("context expired%w", Error)
+func firstError(errs ...error) error {
+	for _, err := range errs {
```

Comment:
I'm not an expert in Go, but it seems like this is only invoked with two arguments. Wouldn't it be cheaper to just hardcode that without using a loop?

Reply:
changed :heavy_check_mark: